crates/telio-proxy/src/proxy.rs

```diff
@@ -345,10 +405,22 @@ impl StateEgress {
                         );
                     }
 
+                    if let Entry::Vacant(e) = self.replaced_sockets.entry(pk) {
+                        let old_socket_addr = self
+                            .sockets
+                            .get(&pk)
+                            .and_then(|(old_socket, _)| old_socket.local_addr().ok());
+                        if let Some(addr) = old_socket_addr {
+                            e.insert(addr);
+                        }
+                    }
                     self.sockets.insert(pk, (socket, None));
+                    self.num_socket_changes += 1;
+                    return SocketResendBehavior::Send;
                 }
             }
         }
+        SocketResendBehavior::NoSend
```

Comment:
NIT: Moving this into match statement would be nice, it's clearer witch path arrives there.